litmus-portal/backend/auth/pkg/manage/manager.go

```diff
@@ -0,0 +1,151 @@
+package manage
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/globalsign/mgo"
+	"golang.org/x/crypto/bcrypt"
+
+	"github.com/litmuschaos/litmus/litmus-portal/backend/auth/pkg/errors"
+	"github.com/litmuschaos/litmus/litmus-portal/backend/auth/pkg/generates"
+	"github.com/litmuschaos/litmus/litmus-portal/backend/auth/pkg/models"
+	"github.com/litmuschaos/litmus/litmus-portal/backend/auth/pkg/store"
+	"github.com/litmuschaos/litmus/litmus-portal/backend/auth/pkg/types"
+)
+
+// NewManager create to authorization management instance
+func NewManager() *Manager {
+	return &Manager{}
+}
+
+// Manager provide authorization management
+type Manager struct {
+	accessGenerate *generates.JWTAccessGenerate
+	userStore      *store.UserStore
+}
+
+// MapAccessGenerate mapping the access token generate interface
+func (m *Manager) MapAccessGenerate(gen *generates.JWTAccessGenerate) {
+	m.accessGenerate = gen
+}
+
+// MustUserStorage mandatory mapping the user store interface
+func (m *Manager) MustUserStorage(stor *store.UserStore, err error) {
+	if err != nil {
+		panic(err)
+	}
+	m.userStore = stor
+	err = m.CreateUser(context.Background(), models.DefaultUser)
+	if err != nil {
+		log.Fatal("Unable to create default user with error:", err)
+	}
+}
+
+// GetUser get the user information
+func (m *Manager) GetUser(ctx context.Context, userName string) (user *models.User, err error) {
+	user, err = m.userStore.GetByUserName(userName)
+	if err != nil && err == mgo.ErrNotFound {
+		err = errors.ErrInvalidUser
+	}
+	return
+}
+
+// CheckUserExists get the user information
+func (m *Manager) CheckUserExists(ctx context.Context, userName string) (bool, error) {
+	_, err := m.GetUser(ctx, userName)
+	if err != nil && err == errors.ErrInvalidUser {
+		return false, nil
+	} else if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// CreateUser get the user information
+func (m *Manager) CreateUser(ctx context.Context, user *models.User) (err error) {

```

Comment:
Can we maintain a consistency, Somewhere i'm seeing the value is getting returned and in someplace we define the variable itself in return and return empty there.